### PR TITLE
PIN-3590 Implemented Archive Agreement Endpoint

### PR DESCRIPTION
**Pull request by nttdata-rtorsoli**

### CODE REVIEW

File: src/main/resources/application.conf [hunk 1]
```diff
@@ -37,6 +37,7 @@ agreement-process {
     user-registry                  = ${USER_REGISTRY_URL}
   }
 
+  events-queue-name         = ${EVENTS_QUEUE_NAME}
```
Comment (beetlecrunch) [inline src/main/resources/application.conf]:
```suggestion
  archiving-purposes-queue-name         = ${ARCHIVING_PURPOSES_QUEUE_NAME}
```
Probably this service will push the same event twice into another queue (I'm talking about the eservice version archiving feature).
So may I suggest renaming like above?

File: src/main/resources/application-standalone.conf [hunk 2]
```diff
@@ -37,6 +37,7 @@ agreement-process {
     user-registry                  = "https://gateway.interop.pdnd.dev/user-registry/0.1"
   }
 
+  events-queue-name         = ${EVENTS_QUEUE_NAME}
```
Comment (beetlecrunch) [inline src/main/resources/application-standalone.conf]: [same](https://github.com/pagopa/interop-be-agreement-process/pull/238/files#r1253046683)

**Review by TonioGela — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala [hunk 3]
```diff
@@ -49,7 +51,8 @@ final case class AgreementApiServiceImpl(
   fileManager: FileManager,
   offsetDateTimeSupplier: OffsetDateTimeSupplier,
   uuidSupplier: UUIDSupplier,
-  queueService: QueueService
+  queueService: QueueService,
+  queueEventsService: QueueService
```
Comment (beetlecrunch) [inline src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala]:
```suggestion
  certifiedMailQueueService: QueueService,
  archivingPurposesQueueService: QueueService
```

File: src/main/scala/it/pagopa/interop/agreementprocess/common/system/ApplicationConfiguration.scala [hunk 4]
```diff
@@ -32,7 +32,7 @@ object ApplicationConfiguration {
 
     ReadModelConfig(connectionString, dbName)
   }
-
-  val certifiedMailQueueName: String = config.getString("agreement-process.certified-mail-queue-name")
+  val eventsQueueName: String          = config.getString("agreement-process.events-queue-name")
```
Comment (beetlecrunch) [inline src/main/scala/it/pagopa/interop/agreementprocess/common/system/ApplicationConfiguration.scala]: [same](https://github.com/pagopa/interop-be-agreement-process/pull/238/files#r1253046683) considarations

File: src/test/resources/application-test.conf [hunk 5]
```diff
@@ -22,6 +22,7 @@ agreement-process {
     party-process                  = "party-process"
   }
 
+  events-queue-name         = "events-queue-name"
```
Comment (beetlecrunch) [inline src/test/resources/application-test.conf]: [same](https://github.com/pagopa/interop-be-agreement-process/pull/238/files#r1253046683)

**Review by beetlecrunch — APPROVED** [review]

File: events/src/main/scala/it/interop/agreementprocess/events/Events.scala [hunk 6]
```diff
@@ -0,0 +1,14 @@
+package it.pagopa.interop.agreementprocess.events
+
+import it.pagopa.interop.commons.utils.SprayCommonFormats.{uuidFormat, offsetDateTimeFormat}
+import spray.json.{DefaultJsonProtocol, RootJsonFormat}
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport
```
Comment (galales) [inline events/src/main/scala/it/interop/agreementprocess/events/Events.scala]: is `akka` required in this case? is `spary` enough?

File: project/Dependencies.scala [hunk 7]
```diff
@@ -148,5 +148,10 @@ object Dependencies {
         pagopa.tenantManagementClient    % Compile,
         scalatest.core                   % Test
       )
+    lazy val events: Seq[ModuleID] =
+      Seq(
+        akka.httpJson                    % Compile,
```
Comment (galales) [inline project/Dependencies.scala]: same for `spray`

File: src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala [hunk 8]
```diff
@@ -1267,4 +1270,43 @@ final case class AgreementApiServiceImpl(
...
+          stamps = agreement.stamps.copy(archiving = Stamp(uid, offsetDateTimeSupplier.get()).some)
+        )
+      )
+    } yield agreement
+
+    val result: Future[Agreement] = for {
+      requesterOrgId <- getOrganizationIdFutureUUID(contexts)
+      agreementUUID  <- agreementId.toFutureUUID
+      agreement      <- agreementManagementService.getAgreementById(agreementUUID)
+      _              <- assertRequesterIsProducer(requesterOrgId, agreement)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala]: As per story description
```suggestion
      _              <- assertRequesterIsConsumer(requesterOrgId, agreement)
```

**Review by galales — CHANGES_REQUESTED** [review]
Minor point on dependencies.
Fixed required on authorization

**Review by galales — APPROVED** [review]

**Pull request merged**